billy-portugal/src/test/java/com/premiumminds/billy/portugal/services/export/webservice/ClientTest.java

```diff
@@ -0,0 +1,703 @@
+import javax.xml.ws.Endpoint;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ClientTest {
+
+    public static final String PRIVATE_KEY_DIR = "/keys/private.pem";
+
+    public static final Integer SOFTWARE_CERTIFICATION = 12;
+
+    private Injector injector;
+    private PTApplication application;
+    private PTInvoice invoice;
+    private PTCreditNote creditNote;
+
+    private StubFatcorewsPort webService;
+
+    private URL stubServerUrl;
+    private Endpoint publish;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        injector = Guice.createInjector(new PortugalDependencyModule(),
+                new PortugalPersistenceDependencyModule("BillyPortugalTestPersistenceUnit"));
+        injector.getInstance(PortugalDependencyModule.Initializer.class);
+        injector.getInstance(PortugalPersistenceDependencyModule.Initializer.class);
+        PortugalBootstrap.execute(injector);
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+
+        BillyPortugal billyPortugal = new BillyPortugal(injector);
+        PTIssuingParams invoiceParameters = getPtInvoiceIssuingParams();
+        PTIssuingParams creditNoteParameters = getPtCreditNoteIssuingParams();
+
+        PTApplication.Builder applicationBuilder = getPTApplicationBuilder(billyPortugal);
+        application = createPtApplication(billyPortugal, applicationBuilder);
+        PTBusiness business = createPtBusiness(billyPortugal, applicationBuilder);
+        PTCustomer customer = createPtCustomer(billyPortugal);
+
+        createSeries(invoiceParameters.getInvoiceSeries(), business, "CCCC2345");
+        createSeries(creditNoteParameters.getInvoiceSeries(), business, "CCCC2346");
+
+        final PTTax flatTax = createFlatTax(billyPortugal);
+
+        PTProduct productNormal = createPtProductNormal(billyPortugal);
+        PTProduct productIntermediate = createPtProductIntermediate(billyPortugal);
+        PTProduct productReduced = createPtProductReduced(billyPortugal);
+        PTProduct productExempt = createPtProductExempt(billyPortugal);
+        PTProduct productFlat = createPtProductFlat(billyPortugal, flatTax);
+        invoice = createPtInvoice(dateFormat, billyPortugal, business, invoiceParameters, customer, productNormal, productIntermediate, productReduced, productExempt, productFlat);
+        creditNote = createPtCreditNote(dateFormat, billyPortugal, business, creditNoteParameters, customer, productNormal, invoice);
+
+        stubServerUrl = new URL("http", "localhost", getFreePort(), "/dummy");
+        final String bindingURI = stubServerUrl.toString();
+
+        webService = new StubFatcorewsPort();
+
+        publish = Endpoint.publish(bindingURI, webService);
+    }
+
+    @AfterEach
+    public void tearDown(){
+        publish.stop();
+    }
+
+    static class LineSummaryKey {
```

Comment:
You defined an identical class in the Client implementation. You could define it in a separate file and then reuse the code in both the client and this test

Reply:
moved to proper class